data/json/terrain.json

```diff
@@ -7020,5 +7020,54 @@
       "ter_set": "t_pit_shallow",
       "items": [ { "item": "rock", "count": 8 }, { "item": "stick", "count": [ 8, 12 ] } ]
     }
+  },
+  {
+    "type": "terrain",
+    "id": "t_shrub_cranberry",
+    "looks-like": "t_shrub_blueberry",
+    "name": "cranberry shrub",
```

Comment:
You also need to place those bushes on the map, preferably on swamps, where cranberries grow in natural habitats. You might look for example how cattail is distributed, but cattail is a `furniture` not `terrain` (which allows to place them on water tiles), so some terrains swapping in `mapgen_functions.cpp` might need to happen.